generator.py

```diff
@@ -78,102 +82,70 @@ def regen_crates(path, conf, level=0):
-    print("")
-    print("[OPTIONS]")
-    print("  -h | --help           Display this message")
-    print("  --gir-path [PATH]     Sets the path of the gir executable to run")
-    print("                        (`./gir/target/release/gir` by default)")
-    print("  --gir-files [PATH]    Sets the path of the gir-files folder")
-    print("                        (`gir-files` by default)")
-    print("  --yes                 Always answer `yes` to any question asked by the script")
-    print("  --no-fmt              If set, this script won't run `cargo fmt`")
+def parse_args(args):
+    parser = argparse.ArgumentParser(description='Helper to regenerate gtk-rs crates using gir.',
+                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
 
+    parser.add_argument('path', nargs="*", default='.',
+                        help='Paths in which to look for Gir.toml files')
+    parser.add_argument('--gir-directory', default=DEFAULT_GIR_DIRECTORY,
+                        help='Path of the gir-files folder')
+    parser.add_argument('--gir-path', default=DEFAULT_GIR_PATH,
+                        help='Path of the gir executable to run')
+    parser.add_argument('--yes', action='store_true',
+                        help=' Always answer `yes` to any question asked by the script')
+    parser.add_argument('--no-fmt', action='store_true',
+                        help='If set, this script will not run `cargo fmt`')
 
-def parse_args(args):
-    conf = {
-        "gir_path": None,
-        "gir_files": None,
-        "yes": False,
-        "run_fmt": True,
-    }
-    i = 0
-
-    while i < len(args):
-        arg = args[i]
-        if arg == "-h" or arg == "--help":
-            print_help()
-            return None
-        elif arg == "--gir-path":
-            i += 1
-            if i >= len(args):
-                print("Expected argument after `--gir-path` option...")
-                return None
-            if not isfile(args[i]):
-                print("`{}` file doesn't exist. Aborting...".format(args[i]))
-                return None
-            conf["gir_path"] = args[i]
-        elif arg == "--gir-files":
-            i += 1
-            if i >= len(args):
-                print("Expected argument after `--gir-files` option...")
-                return None
-            if not isdir(args[i]):
-                print("`{}` folder doesn't exist. Aborting...".format(args[i]))
-                return None
-            conf["gir_files"] = args[i]
-        elif arg == "--yes":
-            conf["yes"] = True
-        elif arg == "--no-fmt":
-            conf["run_fmt"] = False
-        else:
-            print("Unknown argument `{}`.".format(arg))
-            return None
-        i += 1
-    return conf
+    return parser.parse_args()
 
 
 def main():
     gir_path = None
 
     conf = parse_args(sys.argv[1:])
-    if conf is None:
-        return 1
+    print(conf)
```

Comment:
Maybe remove this line? ;)

Reply:
How pedantic of you :angry: :laughing:

Reply:
I like `print`, but too much `print` kills my brain. :p